### DISCUSSION TITLE
WindTerm 2.1.0 Prerelease 1, 2

### DISCUSSION THREAD

**Discussion by kingToolbox** [kingToolbox/WindTerm#79] [General]
# Usage:
### Windows
Please install **[vc_redist.x86](https://aka.ms/vs/16/release/vc_redist.x86.exe)** if WindTerm **failed** to run.
Please use **WindTerm_Win10_Conpty.exe** to support Windows 10 shell. (**Windows 19H1 [v10.0.18309] and later**)

### Linux (Not available in the pre-release stage)
Please use the following command to run the WindTerm after downloading.
``` shell
tar -xzvf WindTerm_2.1.0_Linux_Portable.tar.gz && cd ./WindTerm_2.1.0 && chmod +x ./WindTerm && ./WindTerm
```

### Upgrade
Copy or move the **"profiles"** folder from the **old Windows/Linux version** to the **new Windows/Linux version** and all the sessions and configuration will be restored.

# Note:

Starting from this version, **unless necessary or the last pre-release version**, the Linux version will no longer be provided in the pre-release stage. 

# Improvements:
- [GUI] Added a count label to the filter input box of the tab list menu.
- [FILER] The file size displayed in the status bar is accurate to 2 digit precision, like 1.42GB.

# Bugs:
- [SSH] **The session cannot log in automatically when using pageant authentication. .** #70 
- [GUI] When the shell combobox in the session dialog is switched to admin:cmd, admin:powershell, gitbash, cygwin, etc., the system combobox fails to automatically set the corresponding system type. 

<hr /><em>This discussion was created from the release <a href='https://github.com/kingToolbox/WindTerm/releases/tag/2.1'>WindTerm 2.1.0 Prerelease 1</a>.</em>

**Comment by yuwen123441** [comment]
is Xmodem supported？

**Reply by kingToolbox** [reply]
Thank you for your concern. The X/Y/Z Modem feature is still under development, but it is almost complete and I think it will be released soon. 

Features that have been completed include:
- Send single or multiple files.
- Receive single or multiple files.
- Drag and drop the files to send.
- Automatically open the local files after received.
- Supports the transfer of large files not larger than 4GB.
- When a remote file with the same name exists, operations such as overwriting, ignoring, appending, and renaming are supported. 
- Use the `Cancel` button, or <ctrl+x><ctrl+x><ctrl+x><ctrl+x><ctrl+x>, or <ctrl+c> key to terminate the transmission at any time. 
- Detailed transfer progress and error messages. 
- Fast and stable transmission speed. 

Unfortunately, limited to the windows version of `rz/sz`, `XModem` and `YModem` cannot be supported for the time being, only `ZModem` can be supported. And the transferred file name can only contain ASCII characters, which means that if your file name contains Chinese characters, the file name of the remote file will appear garbled after transmission, but it does not affect the correctness of the file content, so the problem is not too serious. This will be resolved in the subsequent versions. 

The Linux version has no file name garbled problem because it supports `utf-8` encoding by default. `XModem` and `YModem` should also be supported, but I haven't tested it on the Linux version yet, so I'm not sure.

**Reply by kingToolbox** [reply]
Hi, the new version [2.1.0_prerelease_2](https://github.com/kingToolbox/WindTerm/releases/tag/2.1) has been released and supports [ZModem](https://kingtoolbox.github.io/2021/04/23/modem-zmodem/), please download and check, thank you!

**Comment by muyiwhy** [comment]
Is there a plan for MAC version releases?

**Reply by kingToolbox** [reply]
Yes, for sure, as I replied to you in #46 at the beginning of the year. It's just that the progress is delayed a little bit because of the X/Y/Z modem. But it is already in preparation. If nothing happens, the version 2.2.0 to be released in May should be able to include the Mac version. Please keep follow, thank you.

**Comment by Shellfishgene** [comment]
Thanks for focus mode, it seems nice. One small bug I noticed: When WindTerm is maximized (not full screen), and focus mode is turned on with "Full Screen" set to "No", the WindTerm window will be set back to non-maximized. Same when extiting focus mode.

**Reply by kingToolbox** [reply]
Wow, you have already noticed this. :smile:  This problem has just been fixed, but there is still flickering when switching from full screen back to the previous state, and I will continue to try whether it can be eliminated.

**Reply by kingToolbox** [reply]
The new version [WindTerm_2.1.0_Prerelease_6_Windows_Portable](https://github.com/kingToolbox/WindTerm/releases/tag/2.1-prerelease) has just been released. In this version, I have fixed this bug. Unfortunately, I still failed to eliminate the flickering in the end. However, it does not affect daily use. Please download and use it, thank you.

**Comment by Zingsun** [comment]
When uploading and downloading files, ICONS and progress are blurred, like mosaics
